src/components/MarketMySharesForm/index.js

```diff
@@ -359,9 +392,7 @@ class MarketMySharesForm extends Component {
               <th className="marketMyShares__tableHeading marketMyShares__tableHeading--group" />
             </tr>
           </thead>
-          <tbody>
-            {this.generateTableRows()}
-          </tbody>
+          <tbody>{this.generateTableRows()}</tbody>
```

Comment:
This should be also a component, functions should be avoided (for each render we generate a new reference in the stack).

If you need to calculate some values, those calculations should be done in a container of the main component (MarketMySharesForm).

This sub-component should look like:
```
const SharesRows = (firstCalculation, secondCalculation, ...) => <div>
    ... // other necessary html
    { numberOfShares.map( (share) => <div>
        ....
     </div> }
```

Reply:
@apanizo Can you elaborate please? What do you mean with we generate a new reference in the stack?
Afaik the performance hit between a function call and a stateless component is about equal, if not worse. As it would improve our markup, I can still agree with this, I'm just curious about what performance benefits this would gain us, as I've used function calls to render smaller parts of a react component a lot